velox/expression/CastExpr-inl.h

```diff
@@ -249,11 +250,14 @@ VectorPtr CastExpr::applyDecimalToIntegralCast(
   applyToSelectedNoThrowLocal(context, rows, result, [&](int row) {
     auto value = simpleInput->valueAt(row);
     auto integralPart = value / scaleFactor;
-    auto fractionPart = value % scaleFactor;
-    auto sign = value >= 0 ? 1 : -1;
-    bool needsRoundUp =
-        (scaleFactor != 1) && (sign * fractionPart >= (scaleFactor >> 1));
-    integralPart += needsRoundUp ? sign : 0;
+    if (!queryConfig.isCastToIntByTruncate()) {
```

Comment:
The comment for this config at https://github.com/facebookincubator/velox/blob/main/velox/core/QueryConfig.h#L79-L80 looks not right any more. Can we change it to be more general?

Reply:
Updated.